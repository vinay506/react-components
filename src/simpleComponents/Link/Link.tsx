import React from 'react'
import styled from 'styled-components';
function Link(props: any) {
    /**
     * Text: The content which need to be displayed on the link
     * OnLinkClick is an output emitter when the user clicks on the link.
     */
    const { text, onLinkClick, color, fontSize, display } = props;

    const overrideStyles = {
        color: color,
        fontSize: fontSize,
        display: display
    }

    return (
        <React.Fragment>
            <Anchor id="anchor" onClick={onLinkClick} style={overrideStyles}>
                {text}
            </Anchor>
        </React.Fragment>
    )
}

const Anchor = styled.div`
     color: #0275d8;
    cursor: pointer;
    font-size: 16px;
    font-weight: 600;
    `;
export default Link
