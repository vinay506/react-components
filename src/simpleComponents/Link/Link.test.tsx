import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import Link from './Link';
import 'jest-styled-components'

describe('Anchor Link View', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    const callBack = () => {
        return true;
    }
    beforeEach(() => {
        propInfo = {
            text: 'View Schedule',
            onLinkClick: callBack,
            color: 'blue',
            fontSize: '12px',
            display: 'flex'
        };
        defaultStyles = {
            color: '#0275d8',
            cursor: 'pointer',
            fontSize: '16px',
            fontWeight: 600
        }

        wrapper = shallow(<Link {...propInfo} />);
    });
    test('Anchor Link Snapshot', () => {
        const instance = renderer.create(<Link {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Anchor Link Element check', () => {
        const elementLength = wrapper.find('#anchor').length;
        expect(elementLength).toEqual(1);
    })
    test('Anchor Link Styles Validation', () => {
        const elementStyles = wrapper.find('#anchor');
        expect(elementStyles.props().style).toEqual({
            color: propInfo.color, fontSize: propInfo.fontSize, display: propInfo.display
        });
    })

    test('Anchor Link Click Test', () => {
        const button = wrapper.find('#anchor');
        const clickEvent = button.simulate('click');
        const returnValue = expect(clickEvent).toBeCalled;
        expect(returnValue).toBeTruthy();
    })

    // default styles validation
    test('Link Default Width Validation', () => {
        const instance = renderer.create(<Link {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.width);
    })

    test('Link Default cursor Validation', () => {
        const instance = renderer.create(<Link {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.cursor);
    })

    test('Link Default font-size Validation', () => {
        const instance = renderer.create(<Link {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })

    test('Link Default font-weight Validation', () => {
        const instance = renderer.create(<Link {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontWeight);
    })
})
