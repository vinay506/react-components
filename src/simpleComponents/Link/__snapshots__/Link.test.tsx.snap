// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Link View Anchor Link Snapshot 1`] = `
.c0 {
  color: #0275d8;
  cursor: pointer;
  font-size: 16px;
  font-weight: 600;
}

<div
  className="c0"
  id="anchor"
  onClick={[Function]}
  style={
    Object {
      "color": "blue",
      "display": "flex",
      "fontSize": "12px",
    }
  }
>
  View Schedule
</div>
`;
